app/ui/src/main/java/de/mm20/launcher2/ui/launcher/search/location/LocationItem.kt

```diff
@@ -0,0 +1,306 @@
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.ArrowBack
+import androidx.compose.material.icons.rounded.ArrowUpward
+import androidx.compose.material.icons.rounded.Fastfood
+import androidx.compose.material.icons.rounded.Hotel
+import androidx.compose.material.icons.rounded.LocalBar
+import androidx.compose.material.icons.rounded.LocalCafe
+import androidx.compose.material.icons.rounded.LocalGroceryStore
+import androidx.compose.material.icons.rounded.Map
+import androidx.compose.material.icons.rounded.Place
+import androidx.compose.material.icons.rounded.Restaurant
+import androidx.compose.material.icons.rounded.TravelExplore
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import de.mm20.launcher2.search.Location
+import de.mm20.launcher2.search.LocationCategory
+import de.mm20.launcher2.i18n.R
+import de.mm20.launcher2.ui.animation.animateTextStyleAsState
+import de.mm20.launcher2.ui.component.DefaultToolbarAction
+import de.mm20.launcher2.ui.component.Toolbar
+import de.mm20.launcher2.ui.ktx.toPixels
+import de.mm20.launcher2.ui.launcher.search.common.SearchableItemVM
+import de.mm20.launcher2.ui.launcher.search.listItemViewModel
+import de.mm20.launcher2.ui.locals.LocalGridSettings
+import kotlinx.coroutines.launch
+import java.text.DecimalFormat
+import java.time.DayOfWeek
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+import java.time.format.TextStyle
+import java.util.Locale
+import kotlin.math.roundToInt
+
+@Composable
+fun LocationItem(
+    modifier: Modifier = Modifier,
+    location: Location,
+    showDetails: Boolean,
+    onBack: () -> Unit
+) {
+    val context = LocalContext.current
+    val viewModel: SearchableItemVM = listItemViewModel(key = "search-${location.key}")
+    val iconSize = LocalGridSettings.current.iconSize.dp.toPixels()
+
+    val userLocation by viewModel.userLocation.collectAsState(null)
+    val distance = userLocation?.distanceTo(location.toAndroidLocation())
+
+    DisposableEffect(location) {
+        viewModel.init(location, iconSize.toInt())
+        viewModel.startLocationUpdates(context)
+
+        onDispose {
+            viewModel.stopLocationUpdates(context)
+        }
```

Comment:
There should be a lifecycleOwner.repeatOnLifecycle somewhere here to ensure that location updates are stopped when the user navigates away from the launcher (or turns off the screen)

Reply:
I thought that `DisposableEffect` handles this situation by calling `onDispose` when the effect leaves composition. Is that not the case when the screen is turned off? (when the user navigates away?)

Reply:
No, pausing the activity does not destroy the composition. `onDispose` is called when you remove it from the composition but that doesn't happen when you pause the activity.